### PR TITLE
feat: remove `PrimeFieldGmp`

### LINKED ISSUE #77: Remove `PrimeFieldGMP`
### Issue type

Feature Request

### Current behavior?

`PrimeFieldGmp` is provided as a answer when testing correctness of `PriemField` operation. But since `mpz_class` methods are not `constexpr`, we can't run those on GPU.

### Expected Behavior?

Remove `PrimeFieldGmp`.

### PR DESCRIPTION
**Pull request by chokobole**
#Description

`PrimeFieldGmp` was a reference class that was used to check whether our `PrimeField` implement was correct initially. Since the primitives becomes stabilized, we can remove this. By removing this, we can expect to make more things gpu interoperable and computable at compile time.

Resolves #77

### CODE REVIEW

**Review by Insun35 — APPROVED** [review]
LGTM

File: tachyon/math/finite_fields/generator/ext_prime_field_generator/ext_prime_field_generator.cc [hunk 1]
```diff
@@ -38,7 +38,7 @@ int GenerationConfig::GenerateConfigHdr() const {
       "  using BasePrimeField = %{base_prime_field};",
       "  using FrobeniusCoefficient = %{frobenius_coefficient};",
       "",
-      "  // NOTE(chokobole): This can't be constexpr because of PrimeFieldGmp support.",
+      "  // NOTE(chokobole): Make them constexpr.",
```
Comment (TomTaehoonKim) [inline tachyon/math/finite_fields/generator/ext_prime_field_generator/ext_prime_field_generator.cc:41]:
```suggestion
      "  // TODO(chokobole): Make them constexpr.",
```

File: tachyon/math/elliptic_curves/short_weierstrass/test/sw_curve_config.h [hunk 2]
```diff
@@ -1,5 +1,5 @@
-#ifndef TACHYON_MATH_ELLIPTIC_CURVES_SHORT_WEIERSTRASS_TEST_CURVE_CONFIG_H_
-#define TACHYON_MATH_ELLIPTIC_CURVES_SHORT_WEIERSTRASS_TEST_CURVE_CONFIG_H_
+#ifndef TACHYON_MATH_ELLIPTIC_CURVES_SHORT_WEIERSTRASS_TEST_SW_CURVE_CONFIG_H_
```
Comment (TomTaehoonKim) [inline tachyon/math/elliptic_curves/short_weierstrass/test/sw_curve_config.h:1]: I think these changes should come as a separate commit!

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Pull request merged**